main/src/com/google/refine/commands/row/GetRowsCommand.java

```diff
@@ -180,7 +180,8 @@ protected void internalRespond(HttpServletRequest request, HttpServletResponse r
             }
             
             // metadata refresh for row mode and record mode
-	    if (project.getMetadata() != null) {
+	    if (project.getMetadata() != null
+	            && project.getMetadata().getRowCount() == 0) {
```

Comment:
What if the row count changes later on (for instance if the user removes some rows, or uses a transpose operation)? Is the row count going to be updated somewhere else?

I would check if the row count is different from what is stored in the metadata, rather than comparing to zero.

Reply:
The row number should be recounted after operation. Thanks for checking. just make sure.